File: plugin/src/main/java/spp/jetbrains/sourcemarker/PluginUI.java
package spp.jetbrains.sourcemarker;

import com.intellij.ui.Gray;
import com.intellij.ui.JBColor;
import com.intellij.util.ui.UIUtil;

import javax.swing.border.LineBorder;
import java.awt.*;
import java.io.IOException;

public class PluginUI {

    public static final LineBorder PANEL_BORDER = new LineBorder(Gray._85);

    public static Font ROBOTO_LIGHT_BOLD_14;
    private static Font ROBOTO_LIGHT_PLAIN_13;
    private static Font ROBOTO_LIGHT_PLAIN_14;
    private static Font ROBOTO_LIGHT_PLAIN_15;
    private static Font ROBOTO_LIGHT_PLAIN_16;
    private static Font ROBOTO_LIGHT_PLAIN_17;
    private static Font MICROSOFT_YAHEI_PLAIN_12;
    private static Font MICROSOFT_YAHEI_PLAIN_13;
    public static Font MICROSOFT_YAHEI_PLAIN_14;
    private static Font MICROSOFT_YAHEI_PLAIN_15;
    private static Font MICROSOFT_YAHEI_PLAIN_16;

    static {
        try {
            Font ROBOTO_LIGHT = Font.createFont(Font.TRUETYPE_FONT, PluginUI.class.getResourceAsStream("/fonts/Roboto-Light.ttf"));
            ROBOTO_LIGHT_BOLD_14 = ROBOTO_LIGHT.deriveFont(Font.BOLD).deriveFont(14f);
            ROBOTO_LIGHT_PLAIN_13 = ROBOTO_LIGHT.deriveFont(Font.PLAIN).deriveFont(13f);
            ROBOTO_LIGHT_PLAIN_14 = ROBOTO_LIGHT.deriveFont(Font.PLAIN).deriveFont(14f);
            ROBOTO_LIGHT_PLAIN_15 = ROBOTO_LIGHT.deriveFont(Font.PLAIN).deriveFont(15f);
            ROBOTO_LIGHT_PLAIN_16 = ROBOTO_LIGHT.deriveFont(Font.PLAIN).deriveFont(16f);
            ROBOTO_LIGHT_PLAIN_17 = ROBOTO_LIGHT.deriveFont(Font.PLAIN).deriveFont(17f);

            Font YAHEI = Font.createFont(Font.TRUETYPE_FONT, PluginUI.class.getResourceAsStream("/fonts/chinese.msyh.ttf"));
            MICROSOFT_YAHEI_PLAIN_12 = YAHEI.deriveFont(Font.PLAIN).deriveFont(12f);
            MICROSOFT_YAHEI_PLAIN_13 = YAHEI.deriveFont(Font.PLAIN).deriveFont(13f);
            MICROSOFT_YAHEI_PLAIN_14 = YAHEI.deriveFont(Font.PLAIN).deriveFont(14f);
            MICROSOFT_YAHEI_PLAIN_15 = YAHEI.deriveFont(Font.PLAIN).deriveFont(15f);
            MICROSOFT_YAHEI_PLAIN_16 = YAHEI.deriveFont(Font.PLAIN).deriveFont(16f);
        } catch (FontFormatException | IOException e) {
            e.printStackTrace();
        }
    }

    public static final Font BIG_FONT = (PluginBundle.INSTANCE.getLOCALE().getLanguage().equals("zh")) ? MICROSOFT_YAHEI_PLAIN_16 : ROBOTO_LIGHT_PLAIN_17;
    public static final Font SMALL_FONT = (PluginBundle.INSTANCE.getLOCALE().getLanguage().equals("zh")) ? MICROSOFT_YAHEI_PLAIN_15 : ROBOTO_LIGHT_PLAIN_16;
    public static final Font SMALLER_FONT = (PluginBundle.INSTANCE.getLOCALE().getLanguage().equals("zh")) ? MICROSOFT_YAHEI_PLAIN_14 : ROBOTO_LIGHT_PLAIN_15;
    public static final Font SMALLEST_FONT = (PluginBundle.INSTANCE.getLOCALE().getLanguage().equals("zh")) ? MICROSOFT_YAHEI_PLAIN_13 : ROBOTO_LIGHT_PLAIN_14;
    public static final Font SUPER_SMALLEST_FONT = (PluginBundle.INSTANCE.getLOCALE().getLanguage().equals("zh")) ? MICROSOFT_YAHEI_PLAIN_12 : ROBOTO_LIGHT_PLAIN_13;
    public static final Color PANEL_BACKGROUND_COLOR = Gray._37;
    public static final Color LABEL_FOREGROUND_COLOR = new Color(152, 118, 170);
    public static final Color LABEL_FOREGROUND_COLOR1 = new Color(106, 135, 89);
    public static final Color EXPIRY_FOREGROUND_COLOR = Color.decode("#BBBBBB");
    public static final Color SELECT_COLOR_RED = Color.decode("#e1483b");
    public static final Color COMPLETE_COLOR_PURPLE = Color.decode("#9876AA");
    public static final JBColor STATUS_BAR_TXT_BG_COLOR = new JBColor(JBColor.WHITE, Gray._37);
    public static final JBColor CONTROL_BAR_CMD_FOREGROUND = new JBColor(JBColor.BLACK, JBColor.GRAY);
    public static final JBColor DFLT_BGND_COLOR = new JBColor(Gray._242, Gray._50);
    public static final JBColor CNFG_PANEL_BGND_COLOR = new JBColor(Gray._242, Gray._37);
    public static final JBColor BGND_FOCUS_COLOR = new JBColor(Gray._175, Gray._25);
    public static final JBColor COMMAND_TYPE_COLOR = new JBColor(JBColor.BLACK, Gray._125);
    public static final JBColor COMMAND_HIGHLIGHT_COLOR = new JBColor(SELECT_COLOR_RED, Color.decode("#E6E6E6"));

    public static String getCommandTypeColor() {
        return "#" + Integer.toHexString(COMMAND_TYPE_COLOR.getRGB()).substring(2);
    }

    public static String getCommandHighlightColor() {
        return "#" + Integer.toHexString(COMMAND_HIGHLIGHT_COLOR.getRGB()).substring(2);
    }

    public static Color getEditCompleteColor() {
        return UIUtil.getWindowColor();
    }
}

File: common/src/main/kotlin/spp/jetbrains/PluginUI.kt
/*
 * Source++, the open-source live coding platform.
 * Copyright (C) 2022 CodeBrig, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package spp.jetbrains

import com.intellij.ui.Gray
import com.intellij.ui.JBColor
import com.intellij.util.ui.UIUtil
import spp.jetbrains.PluginBundle.LOCALE
import java.awt.Color
import java.awt.Font
import javax.swing.border.LineBorder

object PluginUI {

    @JvmField
    val LABEL_FOREGROUND_COLOR = Color(152, 118, 170)

    @JvmField
    val LABEL_FOREGROUND_COLOR1 = Color(106, 135, 89)

    @JvmField
    val EXPIRY_FOREGROUND_COLOR: Color = Color.decode("#BBBBBB")

    @JvmField
    val SELECT_COLOR_RED: Color = Color.decode("#e1483b")

    @JvmField
    val COMPLETE_COLOR_PURPLE: Color = Color.decode("#9876AA")

    @JvmField
    val STATUS_BAR_TXT_BG_COLOR = JBColor(JBColor.WHITE, Gray._37)

    @JvmField
    val CONTROL_BAR_CMD_FOREGROUND = JBColor(JBColor.BLACK, JBColor.GRAY)

    @JvmField
    val DFLT_BGND_COLOR = JBColor(Gray._242, Gray._50)

    @JvmField
    val CNFG_PANEL_BGND_COLOR = JBColor(Gray._242, Gray._37)

    @JvmField
    val BGND_FOCUS_COLOR = JBColor(Gray._175, Gray._25)

    @JvmField
    val COMMAND_TYPE_COLOR = JBColor(JBColor.BLACK, Gray._125)

    @JvmField
    val COMMAND_HIGHLIGHT_COLOR = JBColor(SELECT_COLOR_RED, Color.decode("#E6E6E6"))

    @JvmStatic
    val commandTypeColor: String
        get() = "#" + Integer.toHexString(COMMAND_TYPE_COLOR.rgb).substring(2)

    @JvmStatic
    val commandHighlightColor: String
        get() = "#" + Integer.toHexString(COMMAND_HIGHLIGHT_COLOR.rgb).substring(2)

    @JvmStatic
    val editCompleteColor: Color
        get() = UIUtil.getWindowColor()

    @JvmField
    val PANEL_BACKGROUND_COLOR: Color = Gray._37

    @JvmField
    val PANEL_BORDER = LineBorder(Gray._85)

    @JvmField
    val ROBOTO_LIGHT_BOLD_14: Font

    private val ROBOTO_LIGHT_PLAIN_13: Font
    private val ROBOTO_LIGHT_PLAIN_14: Font
    private val ROBOTO_LIGHT_PLAIN_15: Font
    private val ROBOTO_LIGHT_PLAIN_16: Font
    private val ROBOTO_LIGHT_PLAIN_17: Font
    private val MICROSOFT_YAHEI_PLAIN_12: Font
    private val MICROSOFT_YAHEI_PLAIN_13: Font

    @JvmField
    val MICROSOFT_YAHEI_PLAIN_14: Font
    private val MICROSOFT_YAHEI_PLAIN_15: Font
    private val MICROSOFT_YAHEI_PLAIN_16: Font

    init {
        try {
            val robotoLightFont = Font.createFont(
                Font.TRUETYPE_FONT,
                PluginUI::class.java.getResourceAsStream("/fonts/Roboto-Light.ttf")
            )
            ROBOTO_LIGHT_BOLD_14 = robotoLightFont.deriveFont(Font.BOLD).deriveFont(14f)
            ROBOTO_LIGHT_PLAIN_13 = robotoLightFont.deriveFont(Font.PLAIN).deriveFont(13f)
            ROBOTO_LIGHT_PLAIN_14 = robotoLightFont.deriveFont(Font.PLAIN).deriveFont(14f)
            ROBOTO_LIGHT_PLAIN_15 = robotoLightFont.deriveFont(Font.PLAIN).deriveFont(15f)
            ROBOTO_LIGHT_PLAIN_16 = robotoLightFont.deriveFont(Font.PLAIN).deriveFont(16f)
            ROBOTO_LIGHT_PLAIN_17 = robotoLightFont.deriveFont(Font.PLAIN).deriveFont(17f)

            val yaheiFont = Font.createFont(
                Font.TRUETYPE_FONT,
                PluginUI::class.java.getResourceAsStream("/fonts/chinese.msyh.ttf")
            )
            MICROSOFT_YAHEI_PLAIN_12 = yaheiFont.deriveFont(Font.PLAIN).deriveFont(12f)
            MICROSOFT_YAHEI_PLAIN_13 = yaheiFont.deriveFont(Font.PLAIN).deriveFont(13f)
            MICROSOFT_YAHEI_PLAIN_14 = yaheiFont.deriveFont(Font.PLAIN).deriveFont(14f)
            MICROSOFT_YAHEI_PLAIN_15 = yaheiFont.deriveFont(Font.PLAIN).deriveFont(15f)
            MICROSOFT_YAHEI_PLAIN_16 = yaheiFont.deriveFont(Font.PLAIN).deriveFont(16f)
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }

    @JvmField
    val BIG_FONT = if (LOCALE.language == "zh") MICROSOFT_YAHEI_PLAIN_16 else ROBOTO_LIGHT_PLAIN_17

    @JvmField
    val SMALL_FONT = if (LOCALE.language == "zh") MICROSOFT_YAHEI_PLAIN_15 else ROBOTO_LIGHT_PLAIN_16

    @JvmField
    val SMALLER_FONT = if (LOCALE.language == "zh") MICROSOFT_YAHEI_PLAIN_14 else ROBOTO_LIGHT_PLAIN_15

    @JvmField
    val SMALLEST_FONT = if (LOCALE.language == "zh") MICROSOFT_YAHEI_PLAIN_13 else ROBOTO_LIGHT_PLAIN_14

    @JvmField
    val SUPER_SMALLEST_FONT = if (LOCALE.language == "zh") MICROSOFT_YAHEI_PLAIN_12 else ROBOTO_LIGHT_PLAIN_13
}
